components/modelregistry/modelregistry.go

```diff
@@ -0,0 +1,84 @@
+// Package modelregistry provides utility functions to config ModelRegistry, an ML Model metadata repository service
+package modelregistry
+
+import (
+	"context"
+	"path/filepath"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/components"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+)
+
+var (
+	ComponentName = "model-registry-operator"
+	Path          = deploy.DefaultManifestPath + "/" + ComponentName + "/overlays/odh"
+)
+
+// Verifies that ModelRegistry implements ComponentInterface.
+var _ components.ComponentInterface = (*ModelRegistry)(nil)
+
+// ModelRegistry struct holds the configuration for the ModelRegistry component.
+// +kubebuilder:object:generate=true
+type ModelRegistry struct {
+	components.Component `json:""`
+}
+
+func (m *ModelRegistry) OverrideManifests(_ string) error {
+	// If devflags are set, update default manifests path
+	if len(m.DevFlags.Manifests) != 0 {
```

Comment:
Isn't it incorrect after https://github.com/opendatahub-io/opendatahub-operator/pull/659 ?

Reply:
I guess the issue is that DevFlags could be nil, is that correct? There are still several components where the len(x.DevFlags.Manifests) check is being used. I suppose the replacement should be:
```
if m.DevFlags != nil && len(m.DevFlags.Manifests) != 0 {
```

Reply:
For those components probably the check is implemented before the function call https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/components/codeflare/codeflare.go#L71

Reply:
Good catch. I must have started the PR before this change went in. Added check in Reconcile like other components.